Miniature ferry: lots of style fixes

sarahzinger opened the pull request:
## Links
* https://miniature-ferry.glitch.me/
* https://glitch.manuscript.com/f/cases/3328426/
* https://glitch.manuscript.com/f/cases/3328468/
* https://glitch.manuscript.com/f/cases/3328464/

## GIF/Screenshots and Changes:
* fixes view -> bug
![Screen Shot 2019-05-01 at 5 33 41 PM](https://user-images.githubusercontent.com/6620164/57044064-4f3e6100-6c37-11e9-9ea7-5d1704b2fc2c.png)

* fixes dropdown behavior for new project pop over (arrow where it should be, popover under the button, popover opens from right/left depending on if you're signed in so it doesn't float off the page):
![Screen Shot 2019-05-01 at 5 37 00 PM](https://user-images.githubusercontent.com/6620164/57044243-f3c0a300-6c37-11e9-96b0-617e0ec267f4.png)
![Screen Shot 2019-05-02 at 9 20 38 AM](https://user-images.githubusercontent.com/6620164/57078241-e0efb200-6cbb-11e9-9c8e-bb40474a0653.png)
![Screen Shot 2019-05-02 at 9 22 01 AM](https://user-images.githubusercontent.com/6620164/57078242-e0efb200-6cbb-11e9-847a-72309b107d3e.png)

* fixes dropdown placement in general
![Screen Shot 2019-05-01 at 5 40 30 PM](https://user-images.githubusercontent.com/6620164/57044398-78abbc80-6c38-11e9-9dbd-e52aeaea602c.png)
![Screen Shot 2019-05-01 at 5 40 45 PM](https://user-images.githubusercontent.com/6620164/57044399-78abbc80-6c38-11e9-840e-a0dd267f0960.png)
![Screen Shot 2019-05-01 at 5 41 05 PM](https://user-images.githubusercontent.com/6620164/57044400-78abbc80-6c38-11e9-9dcd-cb2459925909.png)
![Screen Shot 2019-05-01 at 5 41 51 PM](https://user-images.githubusercontent.com/6620164/57044401-78abbc80-6c38-11e9-9414-ba9793bbe452.png)
![Screen Shot 2019-05-01 at 5 44 02 PM](https://user-images.githubusercontent.com/6620164/57044511-c1637580-6c38-11e9-85da-51c53dca350f.png)


## How To Test:
* to test view -> bug try to add a project to a collection on safari
* look at glitch on a mobile phone or mimic it with the dev tools try to create a new project when signed in/not signed in see if things look right
* test dropdowns on user/team/collection/search pages

## Feedback I'm looking for:
anything I'm missing? See better ways to do what I'm doing that I didn't think about?

## Things left to do before deploying:
- [ ] I mainly tested on safari and chrome, will give this a quick look on edge as well

sarahzinger commented on src/components/header/header.styl:
```diff
@@ -8,6 +8,7 @@
       display: inline-block
       padding-right: 5px
       border: none
+      min-width: 48px
```
this fixes the `view ->` button bug

sheridanvk replied:
Chatting with Sarah this resolves the dropdown caret next to the user signin button

ttseng replied:
nice!

sarahzinger commented on src/components/header/index.js:
```diff
@@ -33,7 +33,7 @@ const Header = ({ searchQuery, showNewStuffOverlay }) => {
         <ul className={styles.buttons}>
           <li className={styles.buttonWrap}>
-            <NewProjectPop />
+            <NewProjectPop opensFromRight={!(currentUser && currentUser.login)} />
```
we'll always open from the right if the user is not logged in because there's a lot more space in the header

sarahzinger commented on styles/results-list.styl:
```diff
@@ -46,6 +46,7 @@
       right: 1em
       top: 1em
       z-index: 1
+      width: 65px     
```
this is also for the "view ->" button

sarahzinger commented on styles/pop-overs.styl:
```diff
@@ -26,7 +26,6 @@ popover-secondary-background()
   cursor: initial
   display: block
   position: absolute
-  top: 55px
```
all of the changes here are to ensure the popover dialogs are not on top of the dropdowns buttons, and to ensure it only opens from the right if it has that class

ttseng commented:
awesome - thanks for id-ing and documenting these changes so thoroughly. 

while you're making these modifications, do you mind also adding this small change?

in `src/components/header/header.styl` add margin-right: 10px to add some space between the logo + search bar on small screens
```
.logoWrap
    flex: 0 0 70px
    margin-right: 10px
```
![image](https://user-images.githubusercontent.com/519336/57086613-77c46a80-6ccc-11e9-9f05-4c60a5686e8a.png)

sarahzinger commented:
@sheridanvk looked into the view -> width thing and realized that the issue is not with the wrapping but that we use `width: fit-content` in our generic popover styles which renders differently depending on safari vs chrome and there's no need for it on this particular thing, so I reset with `width: auto` which fixed it. 

also made a case and updated an existing one with the other thoughts we talked about!
Avatar Dropdown Component Case: https://glitch.manuscript.com/f/cases/3328478/Avatar-dropdown-Component
Added suggestions to the popover component case: https://glitch.manuscript.com/f/cases/3328381/Build-Popover-component
